dev/src/write-batch.ts

```diff
@@ -268,16 +269,27 @@ export class WriteBatch {
     return this;
   }
 
+  set<T>(
+    documentRef: DocumentReference<T>,
+    data: Partial<T>,
+    options: SetOptions
+  ): WriteBatch;
+  set<T>(
+    documentRef: DocumentReference<T>,
+    data: T,
+    options?: SetOptions
+  ): WriteBatch;
+
   /**
    * Write to the document referred to by the provided
-   * [DocumentReference]{@link DocumentReference}.
-   * If the document does not exist yet, it will be created. If you pass
-   * [SetOptions]{@link SetOptions}., the provided data can be merged
-   * into the existing document.
+   * [DocumentReference]{@link DocumentReference}. If the document does not
+   * exist yet, it will be created. If you pass [SetOptions]{@link SetOptions},
+   * the provided data can be merged into the existing document. Using Partial
+   * objects requires [SetOptions]{@link SetOptions} to be passed in as well.
```

Comment:
Same comment regarding refocusing this on "merge".

Reply:
done.